opsdroid/connector/matrix/connector.py

```diff
@@ -224,18 +237,24 @@ async def get_nick(self, roomid, mxid):
         """
         if self.room_specific_nicks:
             try:
-                return await self.connection.get_room_displayname(roomid, mxid)
+                res = await self.connection.get_displayname(mxid)
```

Comment:
Here it should be a room specific name. Right now its just returning the default username.

Reply:
I'm sorry I can't find anything in the matrix spec how room specific displaynames can be set or read. Can you point me to the corresponding part of the documentation?

Reply:
I dont think there's a direct way to do this. Seems like you'll have to match the mxid with the usernames in the list returned by [this function](https://matrix-nio.readthedocs.io/en/latest/nio.html?highlight=members#nio.AsyncClient.joined_members) and get the corresponding displayname.